### ISSUE TITLE
Transformation potion has wong name

### ISSUE THREAD

**Issue by zR1OQicz** [Innoxia/liliths-throne-public#1240]
**Describe the bug**
Each Imp group in Submission's tunnels has a transformation potion. This potion works, but has the name of a different item. Usually the last one I have enchanted myself.

**Expected behavior**
The generated potion has a correct name.

**To Reproduce**
1. Enchant some item (e.g. a pair of shoes)
2. Go down to Submission
3. Enter the Imp tunnels
4. Fight and defeat a group of Imps
5. Search their inventories to find a transformation potion which has the same name as the enchanted shoes.

**Attach files**
error.log
Version: 0.3.5.4
Java: 13
Warning: Nashorn engine is planned to be removed from a future JDK release
var timer not found...
Warning: Nashorn engine is planned to be removed from a future JDK release
var timer not found...
var timer not found...

console output
(java:7163): dbind-WARNING **: 21:22:00.930: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-Gkkp5pUfCh: Connection refused
Printing to error.log

(java:7163): Gdk-WARNING **: 21:22:10.788: XSetErrorHandler() called with a GDK error trap pushed. Don't do that.

**Additional information:**
 - Game version  	v0.3.5.4
 - OS: SuSE Linux 64-bit
 - self-compiled version from cloned github source
- java --version
openjdk 13 2019-09-17
OpenJDK Runtime Environment (build 13+33-suse-1.1-x8664)
OpenJDK 64-Bit Server VM (build 13+33-suse-1.1-x8664, mixed mode)

**Comment by zR1OQicz** [comment]
Proposed fix: re-generate the potion name after automatically creating it

```
diff --git a/src/com/lilithsthrone/game/character/npc/submission/ImpAttacker.java b/src/com/lilithsthrone/game/character/npc/submission/ImpAttacker.java
index 04bcac83..32bd83ae 100644
--- a/src/com/lilithsthrone/game/character/npc/submission/ImpAttacker.java
+++ b/src/com/lilithsthrone/game/character/npc/submission/ImpAttacker.java
@@ -363,12 +363,14 @@ public class ImpAttacker extends NPC {
                                Value<AbstractItemType, Map<ItemEffect, String>> effects = TunnelImpsDialogue.getImpLeader().generateTransformativePotion(Main.game.getPlayer());
                                if(effects!=null) {
                                        AbstractItem potion = EnchantingUtils.craftItem(AbstractItemType.generateItem(effects.getKey()), new ArrayList<>(effects.getValue().keySet()));
+                                       potion.setName(EnchantingUtils.getPotionName(AbstractItemType.generateItem(effects.getKey()), new ArrayList<>(effects.getValue().keySet())));
                                        TunnelImpsDialogue.getImpGroup().get(1).addItem(potion, false);
                                }
                                if(!Main.game.getPlayer().getNonElementalCompanions().isEmpty()) {
                                        Value<AbstractItemType, Map<ItemEffect, String>> effects2 = TunnelImpsDialogue.getImpLeader().generateTransformativePotion(Main.game.getPlayer().getMainCompanion());
                                        if(effects2!=null) {
                                                AbstractItem potion2 = EnchantingUtils.craftItem(AbstractItemType.generateItem(effects2.getKey()), new ArrayList<>(effects2.getValue().keySet()));
+                                               potion2.setName(EnchantingUtils.getPotionName(AbstractItemType.generateItem(effects.getKey()), new ArrayList<>(effects.getValue().keySet())));
                                                TunnelImpsDialogue.getImpGroup().get(1).addItem(potion2, false);
                                        }
                                }
```

**Comment by zR1OQicz** [comment]
Verified fixed with v0.3.7.3
